worlds/doom_1993/__init__.py

```diff
@@ -0,0 +1,185 @@
+            "Warrens - Rocket launcher",
+            "Warrens - Rocket launcher 2",
+            "Warrens - Partial invisibility",
+            "Warrens - Invulnerability",
+            "Warrens - Supercharge",
+            "Warrens - Berserk",
+            "Warrens - Chaingun"
+        ])
+    
+
+    def place_locked_item_in_locations(self, item_name, locations):
+        location = self.multiworld.random.choice(locations)
+        self.multiworld.get_location(location, self.player).place_locked_item(self.create_item(item_name))
+
+
+    def generate_early(self):
+        # Give starting levels right away
+        for starting_item in self.starting_items:
+            self.multiworld.push_precollected(self.create_item(starting_item))
+  
+            
+    def generate_basic(self) -> None:
+        self.multiworld.completion_condition[self.player] = lambda state: self.has_achieved_final_goal(state)
+        
+    
+    def has_achieved_final_goal(self, state):
+        for event in Events.events:
+            if not state.has(event, self.player):
+                return False
+        return True
+    
+    
+    def create_item(self, name: str) -> DOOM1993Item:
+        item_id: int = self.item_name_to_id[name]
+        doom_1993_item = DOOM1993Item(name, Items.item_table[item_id]["classification"], item_id, player=self.player)
+        self.item_name_to_item[name] = doom_1993_item
+        return doom_1993_item
+    
+    
+    def create_item_with_classification(self, name: str, classification: ItemClassification) -> DOOM1993Item:
+        item_id: int = self.item_name_to_id[name]
+        doom_1993_item = DOOM1993Item(name, classification, item_id, player=self.player)
+        self.item_name_to_item[name] = doom_1993_item
+        return doom_1993_item
+
+
+    def create_items(self):
+        # Items
+        for item_id, item in Items.item_table.items():
+            count = item["count"]
+            if str(item["name"]) in self.starting_items:
+                count -= 1
+            for i in range(0, count):
+                classification = item["classification"]
+                if i < item["progression_count"]:
+                    classification = ItemClassification.progression
+                doom_1993_item = self.create_item_with_classification(item["name"], classification)
+                # See create_regions
+                if item["name"] != "Warrens - Blue skull key" and \
+                   item["name"] != "Halls of the Damned - Yellow skull key":
+                    self.multiworld.itempool += [doom_1993_item]
+
+
+    def fill_slot_data(self) -> Dict[str, Any]:
+        slot_data = {}
+
+        for option_name in self.option_definitions:
+            option = getattr(self.multiworld, option_name)[self.player]
+            slot_data[option_name] = option.value
+
+        # So it works with APCpp, it's looking for "DeathLink" instead of "death_link" option used in most games.
```

Comment:
this should prob be a bug ticket for APCpp as `death_link` is the prefered writing